testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/KeycloakContainerFeaturesController.java

```diff
@@ -86,20 +86,32 @@ public UpdateFeature(Profile.Feature feature, boolean skipRestart, FeatureAction
          *
          */
         private void assertValid() {
+            if (shouldSkipValidation())
+                return;
+
             assertThat("An annotation requested to " + action.name()
                             + " feature " + feature.name() + " however it was already in that state" ,
                     ProfileAssume.isFeatureEnabled(feature),
                     is(!(action == FeatureAction.ENABLE)));
         }
 
         private void assertPerformed() {
+            if (shouldSkipValidation())
+                return;
+
             assertThat("An annotation requested to " + action.name() +
                             " feature " + feature.name() + ", however after performing this operation " +
                             "the feature is not in desired state" ,
                     ProfileAssume.isFeatureEnabled(feature),
                     is(action == FeatureAction.ENABLE));
         }
 
+        private boolean shouldSkipValidation() {
+            // KEYCLOAK-12895 - Some features might be in product disabled and in project enabled.
+            // In this case, the current state of the feature can be different.
+            return feature.hasDifferentProductType() && Profile.getName().equals("product");
```

Comment:
I see there are failing tests on travis. Maybe the check for "Profile.getName().equals("product")" should be removed from this line and the check should be always disabled to have the tests passing?

Reply:
Ok, at this time, it could be removed.